packages/app/components/checkout/checkout-return.tsx

```diff
@@ -0,0 +1,124 @@
+import { useEffect, useState, useCallback } from "react";
+
+import { Button } from "@showtime-xyz/universal.button";
+import { useRouter } from "@showtime-xyz/universal.router";
+import { Spinner } from "@showtime-xyz/universal.spinner";
+import { Text } from "@showtime-xyz/universal.text";
+import { View } from "@showtime-xyz/universal.view";
+
+import { useConfirmPayment } from "app/hooks/api/use-confirm-payment";
+import { usePaymentsManage } from "app/hooks/api/use-payments-manage";
+
+import { stripePromise } from "./stripe";
+
+const REDIRECT_SECONDS = 5;
+
+export const CheckoutReturn = () => {
+  const router = useRouter();
+  const [time, setTime] = useState(REDIRECT_SECONDS);
+  const { setPaymentByDefault } = usePaymentsManage();
+  const { paymentStatus, message, confirmPaymentStatus } = useConfirmPayment();
+  const handlePaymentSuccess = useCallback(async () => {
+    setTimeout(() => {
+      router.replace("/drop/free?checkoutSuccess=true");
+    }, REDIRECT_SECONDS * 1000);
+    setInterval(() => {
+      setTime((time) => (time > 0 ? time - 1 : 0));
+    }, 1000);
+
+    const setAsDefaultPaymentMethod = new URLSearchParams(
+      window.location.search
+    ).get("setAsDefaultPaymentMethod");
+
+    const stripe = await stripePromise;
+
+    const clientSecret = new URLSearchParams(window.location.search).get(
+      "payment_intent_client_secret"
+    );
```

Comment:
Same issue as pointed out before. Do we already want to make this future proof and define our components as web-only?

Reply:
URLSearchParams are also not available without Polyfill

Reply:
will do! good catch 🙏